src/aave-v2/lens/RatesLens.sol

```diff
@@ -359,7 +425,49 @@ abstract contract RatesLens is UsersLens {
         );
     }
 
-    /// INTERNAL ///
+    /// @notice Computes and returns the underlying pool rates for a specific market.
+    /// @param _underlying The underlying pool market address.
+    /// @param _supplied The amount hypothetically supplied.
+    /// @param _borrowed The amount hypothetically borrowed.
+    /// @param _repaid The amount hypothetically repaid.
+    /// @param _withdrawn The amount hypothetically withdrawn.
+    /// @return poolSupplyRate The market's pool supply rate per year (in ray).
+    /// @return poolBorrowRate The market's pool borrow rate per year (in ray).
+    function _getPoolRatesPerYear(
+        address _underlying,
+        uint256 _supplied,
+        uint256 _borrowed,
+        uint256 _repaid,
+        uint256 _withdrawn
+    ) internal view returns (uint256 poolSupplyRate, uint256 poolBorrowRate) {
+        DataTypes.ReserveData memory reserve = pool.getReserveData(_underlying);
+
+        PoolRatesVars memory vars;
+        (
+            vars.availableLiquidity,
+            vars.totalStableDebt,
+            vars.totalVariableDebt,
+            ,
+            ,
+            ,
+            vars.avgStableRate,
+            ,
+            ,
+
+        ) = dataProvider.getReserveData(_underlying);
+        (, , , , vars.reserveFactor) = reserve.configuration.getParamsMemory();
+
+        (poolSupplyRate, , poolBorrowRate) = IReserveInterestRateStrategy(
+            reserve.interestRateStrategyAddress
+        ).calculateInterestRates(
+            _underlying,
+            vars.availableLiquidity + _supplied + _repaid - _borrowed - _withdrawn,
```

Comment:
Checking just supply flow. Shouldn't `_repaid` go to a P2P deal instead of adding to pool liquidity and changing pool rates?

Reply:
It's actually what was missing in yearn's implementation: when matching peer-to-peer, Morpho repays the debt borrowed by borrowers matched peer-to-peer, so it effectively has an impact on the pool's rates

Reply:
Oups, misunderstood your remark
Indeed, according to the LendingPool's behavior crossed with ReserveLogic's, upon repaying, debt is first decreased by burning the debt token then interest rates are updated via `updateInterestRates`, inside which the liquidity is increased by the amount repaid
So I did this here too by increasing the liquidity by the amount repaid while also decreasing the debt by the amount repaid

Ref:
https://github.com/aave/protocol-v2/blob/0829f97c5463f22087cecbcb26e8ebe558592c16/contracts/protocol/lendingpool/LendingPool.sol#L269-L277
https://github.com/aave/protocol-v2/blob/6f57232358af0fd41d9dcf9309d7a8c0b9aa3912/contracts/protocol/lendingpool/DefaultReserveInterestRateStrategy.sol#L130-L142